### ISSUE TITLE
Missing Converter handling for full class name=java.util.UUID, simple name=UUID>

### ISSUE THREAD

**Issue by stone-wlg** [elastic/logstash#8982]
- Version: 6.1.2
- Operating System: Ubuntu 16.04
Hi all,
    when i use logstash.inputs.jdbc to read a UUID column, it throws exception. i read source code, i seems that it does not matter about logstash.inputs.jdbc plugin, it is about Logstash Event does not support UUID.
[2018-01-18T15:12:03,245][WARN ][logstash.inputs.jdbc     ] Exception when executing JDBC query {:exception=>#<Sequel::DatabaseError: Java::OrgLogstash::MissingConverterException: Missing Converter handling for full class name=java.util.UUID, simple name=UUID>}

**Comment by guyboertje** [comment]
@stone-wlg 

The Event class in Logstash only supports a limited number of datatypes. This is down to use needing to serialize the Event to the Persistent Queue and to JSON.

The JDBC input does not have the ability to do all conversions from the many different SQL datatypes (across the different db technologies and jdbc driver implementations) to the limited set of LS Event datatypes.
 
You should use your DB technology's casting functions to change the uuid type to a varchar.

**Comment by stone-wlg** [comment]
@guyboertje 
Thanks for your reply, but logstash 5.4.0 works, it seems logstash code has been upgraded.

**Comment by jacobCRP** [comment]
We're having this same exact problem. Something changed in a new version of LS.

**Comment by jacobCRP** [comment]
Any updates with this?

**Comment by josemanuelfuentes** [comment]
I have been reading a lot of entries on forums suggesting the casting as @guyboertje suggests. 
Now is it possible to use a filter like Grok to change the cast. for example my field in a postgres table is INET can I use grok filter and type it to 
IPV4 (?<![0-9])(?:(?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5])[.](?:[0-1]?[0-9]{1,2}|2[0-4][0-9]|25[0-5]))(?![0-9])

In other words do I still need the CAST on the sql and then use Grok? Or can Grok take care of the cast?
Thanks!
Any link to use of grok with this intent would be welcome.

**Comment by guyboertje** [comment]
Yes you need to CAST in the SQL statement. The error occurs in the jdbc input when the "incompatible" datatype is processed at the time the event/doc is being created.
